### PR TITLE
VAN-2123 Update deployment templates to support secret values

### PR DESCRIPTION
**Pull request by pranaySinghDev**
Use underlying secret managment objects to hide sensetive variables from users
* VAN-2288 Added client service account to cloud run 
* Added secret manager to cloud run template
* VAN-2219 Added k8s secret object for basic-gke helm chart
* VAN-2219 Added k8s secret object for istio-gke helm chart
* VAN-2075 Added jinja variable for application name

### CODE REVIEW

**Comment by pranaySinghDev** [comment]
@dshelley66 Fixed discovered a weird bug in yaml
Apparently, knative cloud run specs treat double quotes and single quotes in yaml differently now.

**Comment by dshelley66** [comment]
ok I've tested the updated PR with serverless and CDN and it looks good. Wondering if it is worth running the test suite against this? @rajatsharma111 is that possible?

**Comment by rajatsharma111** [comment]
@dshelley66 @pranaySinghDev I have run the test with these pipeline-modules changes and found a couple of issues-

1- ERROR: (gcloud.secrets.delete) PERMISSION_DENIED: Secret Manager API has not been used in project 196929166347 before or it is disabled. Enable it by visiting https://console.developers.google.com/apis/api/secretmanager.googleapis.com/overview?project=196929166347 then retry. If you enabled this API recently, wait a few minutes for the action to propagate to our systems and retry.

**Cloud Build URL**- [https://console.cloud.google.com/cloud-build/builds;region=global/62262e82-31b4-480e-873f-3f9a81d53f0a?project=vanguard-api-automation-test](https://console.cloud.google.com/cloud-build/builds;region=global/62262e82-31b4-480e-873f-3f9a81d53f0a?project=vanguard-api-automation-test)

**Resolution**- Gave **Secret Manager Secret Accessor** permission to the automation service account(codepipes-qa@vanguard-api-automation-test.iam.gserviceaccount.com)

After changing the permission we triggered the test again and got the new error-

2- {"id":"a4fbf1b6-6343-403d-b623-fee216fce640", "message":"An unexpected error occurred: build has finished with status 'Failure': 'FAILURE' in 'gcr.io/cloud-builders/gcloud bash -c echo \"\napiVersion: serving.knative.dev/v1\nkind: Service\nmetadata:\n  name: autoappqpesr\nspec:\n  template:\n    metadata:\n      labels:\n        codepipes-project: autoproject-1641970432620985000\n        codepipes-environment: autoenvironment-1641970440014725000\n    spec:\n      containers:\n      - image: gcr.io/vanguard-api-automation-test/cloudrun-server:latest\n        ports:\n        - name: http1\n          containerPort: 8080\n        env:\" >> app.yaml\ncat /workspace/.env | while IFS= read -r line; do\n  value=${line#*=}\n  name=${line%%=*}\n  printf '        - name: %s\\n          value: %s\\n' \"$name\" \"$value\" >>  app.yaml\ndone;\ncat /workspace/.secret.env | while IFS= read -r line; do\n  value=${line#*=}\n  name=${line%%=*}\n  printf \"        - name: %s\\n          valueFrom:\\n            secretKeyRef:\\n              key: '1'\\n              name: %s\\n\" \"$name\" \"autoappqpesr-$name\" >>  app.yaml\ndone;\ngcloud run services replace app.yaml --region us-central1\n'", "depId":"6712d929-5f7c-4a5b-9d68-bd973f75073a", "pipelineStatus":"Failure", "createdAt":"1641970819", "updatedAt":"1641970954", "jobStatus":"failed", "buildId":"8a3597d6-82f0-4938-93c5-3e13ae9fa923", "jobType":"deploy", "summary":{}, "totalSteps":2, "progress":1, "steps":[{"name":"Deploy", "description":"Trigger the deployment"}, {"name":"Sync", "description":"Wait for the completion and update the status"}]}

**Cloud Build URL**- [https://console.cloud.google.com/cloud-build/builds;region=global/8a3597d6-82f0-4938-93c5-3e13ae9fa923;step=3?project=vanguard-api-automation-test](https://console.cloud.google.com/cloud-build/builds;region=global/8a3597d6-82f0-4938-93c5-3e13ae9fa923;step=3?project=vanguard-api-automation-test)

**Comment by dshelley66** [comment]
@kanaksinghal Since @rajatsharma111 found some issues I was expecting that he would re-run the testing.

**Comment by rajatsharma111** [comment]
@pranaySinghDev @kanaksinghal I have tested this PR for voting(AWS), apprunner(AWS)
, cloud run(GCP) and cdn(GCP) app, all works fine.

**Review by kanaksinghal — APPROVED** [review]

File: pipeline-modules/deployment-service/gcp/cloud-run.yaml [hunk 1]
```diff
@@ -74,34 +74,43 @@ template: |
...
-        name: {{applicationName|lower}}
+        name: {{application_name}}
       spec:
         template:
           metadata:
             labels:
               codepipes-project: {{project}}
               codepipes-environment: {{environment}}
           spec:
+            serviceAccountName: $$client_email
```
Comment (pranaySinghDev) [inline pipeline-modules/deployment-service/gcp/cloud-run.yaml:113]: @kanaksinghal Here is VAN-2288 i'll modify the commit message

**Review by rajatsharma111 — APPROVED** [review]

**Pull request merged**